analyzer/runtime/runtime.go

```diff
@@ -74,8 +74,8 @@ func NewRuntimeAnalyzer(
 	}, nil
 }
 
-func (m *Main) Start() {
-	ctx := context.Background()
+func (m *Main) Start(ctx context.Context) {
+	defer m.cleanup()
 
 	if err := m.prework(); err != nil {
```

Comment:
Also pass the `ctx` into `prework` so that this cancelable context is also used in `SendBatch`.

Reply:
thanks, are there any other context.Background's around?